### PR TITLE
Update Demo Example App

### PR DESCRIPTION
**Pull request by vdolbilov**
Addresses #180 

Hi, I updated the demo to RN 0.45.1. A few notes:

- Couldn't update RN via `react-native upgrade` or `react-native-git-upgrade` so just nuked the demo and created a new RN project.
- In the demo, added `getOrientation` and `getSpecificOrientation` button to IOS and `getOrientation` to android
- Added lock to left and right on android
- Changed where in the component the Orientation calls are made to match the example at the bottom of the README
- It seems that lockToLandscapeRight and Left are actually reversed? On both IOS and android. Tap lock to left on IOS and the specific orientation will actually be `LANDSCAPE-RIGHT`

### CODE REVIEW

File: demo/index.android.js [hunk 1]
```diff
@@ -9,18 +10,32 @@ import {
...
+class Demo extends Component {
   constructor() {
     super();
-    const init = Orientation.getInitialOrientation();
-    this.state = {
-      init,
-      or: init,
-    };
+
     this._updateOrientation = this._updateOrientation.bind(this);
```
Comment (andrerfneves) [inline demo/index.android.js]: Shouldn't need to `bind(this)` if you use an Arrow Function directly on the `_updateOrientation` method

File: demo/index.android.js [hunk 2]
```diff
@@ -9,18 +10,32 @@ import {
...
+    Orientation.removeOrientationListener(this._updateOrientation);
+  }
+
+  _getOrientation() {
+    Orientation.getOrientation((err, orientation) => {
+      Alert.alert(`Orientation is ${orientation}`);
+    });
+  }
+
   _updateOrientation(or) {
```
Comment (andrerfneves) [inline demo/index.android.js]: `_updateOrientation = (or) => { ...`

File: demo/index.android.js [hunk 3]
```diff
@@ -9,18 +10,32 @@ import {
```
Comment (andrerfneves) [inline demo/index.android.js]: Unnecessary empty space

File: demo/index.ios.js [hunk 4]
```diff
@@ -9,21 +10,41 @@ import {
```
Comment (andrerfneves) [inline demo/index.ios.js]: Unnecessary empty line

File: demo/index.ios.js [hunk 5]
```diff
@@ -9,21 +10,41 @@ import {
...
   constructor() {
     super();
-    const init = Orientation.getInitialOrientation();
-    this.state = {
-      init,
-      or: init,
-      sor: init,
-    };
+
     this._updateOrientation = this._updateOrientation.bind(this);
```
Comment (andrerfneves) [inline demo/index.ios.js]: Same `binding` comment as above on the Android app code. Use Arrow Functions instead

File: demo/package.json [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
-  "private": true,
-  "scripts": {
-    "start": "node_modules/react-native/packager/packager.sh"
-  },
-  "dependencies": {
-    "react": "^15.1.0",
-    "react-native": "^0.27.2",
-    "react-native-orientation": ".."
-  }
+	"name": "demo",
```
Comment (andrerfneves) [inline demo/package.json]: Let's check the tabbing/indentation in this package.json file to set it correctly. Also let's maybe change the name from `demo` to `ReactNativeOrientationDemo` or something more descriptive.

File: demo/package.json [hunk 7]
```diff
@@ -1,13 +1,16 @@
...
+	"version": "2.0.0",
+	"private": true,
+	"scripts": {
+		"start": "node node_modules/react-native/local-cli/cli.js start"
+	},
+	"dependencies": {
+		"react": "16.0.0-alpha.12",
+		"react-native": "^0.45.1",
+		"react-native-orientation": "file:n.."
```
Comment (andrerfneves) [inline demo/package.json]: Also, what does `file:n..` represent here?

**Comment by andrerfneves** [comment]
@vdolbilov Great stuff! Thanks for taking the initiative to update the Demo. I added of minor comments to the code commits you made. If you could check those out that'd be great. The idea really was to nuke the old demo for a brand new one since it was quite outdated. 

P.S. I'm going to test your demo in a few minutes when I get back to my machine and I'll be able to give more detailed description as to why LEFT and RIGHT are flipped. But if that's the case, it should be a quick fix. Did you find that on both the Android and iOS builds @vdolbilov?

**Comment by stoneman1** [comment]
Nice work! Check what @andrerfneves said and then we could merge it?

File: demo/package.json (hunk 7 above)
Comment (vdolbilov) [inline demo/package.json]: This was a typo, it should have been just `file:..`. Specifies the `react-native-orientation` dependency as a file path.

**Comment by vdolbilov** [comment]
@andrerfneves Thanks for the review, I fixed the things you mentioned (at least I think I did about the whitespace). 

I did some more testing of the LEFT RIGHT and it looks like it only happens when you explicitly call lockToLandscapeRight or lockToLandscapeLeft on the IOS simulator. At least, the specific orientation text in the state is a mismatch. It looks like unlocking all orientations and then rotating the emulator works as expected. On android I think it's working correctly.

**Comment by andrerfneves** [comment]
@vdolbilov I'm going to check this again and merge it today. Re: the `lockToLandscapeRight/left` on iOS simulator I will try and test this out with this example in a couple devices I have available. If the issue is real we'll add an issue for it and track it there. Thanks

**Pull request merged**